src/ILCompiler.Compiler/src/IL/ILImporter.Scanner.cs

```diff
@@ -184,6 +184,17 @@ private void StartImportingInstruction()
         {
             _previousInstructionOffset = _currentInstructionOffset;
             _currentInstructionOffset = _currentOffset;
+            _isPrefixInstruction = false;
+        }
+
+        private void EndImportingInstruction()
+        {
+            // If we have any pending prefixes and the instruction we just processed is not a prefix instruction,
+            // it consumed the prefix. Clear the state.
+            if (!_isPrefixInstruction)
```

Comment:
This check for `_isPrefixInstruction` should not be necessary. EndImportingInstruction is not called after the prefix, it is only called after the whole instruction is imported.